test/rust-test-util/TestResultReportingClient/src/models/test_result.rs

```diff
@@ -0,0 +1,106 @@
+use std::fmt::{self, Formatter};
+
+use serde::{
+    de::{self, Unexpected, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TestOperationResultDto {
+    #[serde(rename = "source")]
+    source: String,
+    #[serde(rename = "result")]
+    result: TestResult,
+    #[serde(rename = "type")]
+    _type: String,
```

Comment:
We should probably make this an enum. I believe appropriate values are 0 to ~10.